event-sourcing/subscription-manager/src/main/java/uk/gov/justice/services/event/sourcing/subscription/jmx/MBeanInstantiator.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.services.event.sourcing.subscription.jmx;
+
+
+import static java.lang.management.ManagementFactory.getPlatformMBeanServer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+@Singleton
+@Startup
+public class MBeanInstantiator {
+
+    private static final String SHUTTERING_DOMAIN_NAME = "shuttering";
+    private static final String CATCHUP_DOMAIN_NAME = "catchup";
+
+    private static final String SHUTTERING_BEAN = "DefaultShutteringMBean";
+    private static final String CATCHUP_BEAN = "DefaultCatchupMBean";
+
+    private Map<Object, ObjectName> mbeanMap = createMBeanMap();
+    private MBeanServer mbeanServer = getPlatformMBeanServer();
+
+    @PostConstruct
+    public void registerMBeans() {
+        mbeanMap.forEach((key, value) -> {
+            try {
+                mbeanServer.registerMBean(key, value);
+            } catch (final Exception e) {
+                throw new MBeanException(String.format("Error during registration MXBean registration: %s", e.getMessage()));
+            }
+        });
+    }
+
+    @PreDestroy
+    public void unregisterMBeans() {
+        mbeanMap.forEach((key, value) -> {
+            try {
+                mbeanServer.unregisterMBean(value);
+            } catch (final Exception e) {
+                throw new MBeanException(String.format("Error during MXBean unregistration: %s", e.getMessage()));
+            }
+        });
+    }
+
+    private Map<Object, ObjectName> createMBeanMap() {
+        try {
+            final Map<Object, ObjectName> mbeanMap = new HashMap<>();
+
+            mbeanMap.put(DefaultShutteringMBean.class.newInstance(), new ObjectName(SHUTTERING_DOMAIN_NAME, "type", SHUTTERING_BEAN));
+            mbeanMap.put(DefaultCatchupMBean.class.newInstance(), new ObjectName(CATCHUP_DOMAIN_NAME, "type", CATCHUP_BEAN));
+
+            return mbeanMap;
+        } catch (final Exception exception) {
```

Comment:
Might be better to catch the actual Exceptions that are thrown rather than this catch all